common/lzma/Alloc.c

```diff
@@ -280,13 +286,14 @@ const ISzAlloc g_BigAlloc = { SzBigAlloc, SzBigFree };
 */
 #define MY_ALIGN_PTR_DOWN(p, align) ((void *)((((UIntPtr)(p)) & ~((UIntPtr)(align) - 1))))
 
-#define MY_ALIGN_PTR_UP_PLUS(p, align) MY_ALIGN_PTR_DOWN(((char *)(p) + (align) + ADJUST_ALLOC_SIZE), align)
 
-/*
-#if (_POSIX_C_SOURCE >= 200112L) && !defined(_WIN32)
+#if !defined(_WIN32) && !defined(_EE) && defined(_POSIX_C_SOURCE) && (_POSIX_C_SOURCE >= 200112L)
```

Comment:
Is really needed the `!defined(_EE)` here?
If I'm right we always compile without the `_EE` flag

Reply:
The `_EE` macro comes from `EE_CPPFLAGS` at file `stub/lzma/Makefile` (as done in other stubs)

https://github.com/ps2dev/ps2-packer/blob/5a6f491cba23fcaa13ab690997d9f24fe8892641/stub/lzma/Makefile#L7

As this problem is still present in the latest LZMA SDK, instead of commenting out the `USE_posix_memalign` macro (thus disabling the use of [`posix_memalign(3)`](https://man7.org/linux/man-pages/man3/posix_memalign.3.html) in any non-windows platform) as done in commit [e3ceaec](/ps2dev/ps2-packer/commit/e3ceaec5b760f8dd49ae92b6290a35cd43979d28#diff-7df6eb997f8e3984df7abdb9e3afb55bd3e883dd263c68a066c7a219bf3ddcb6), I opted to use the `_EE` macro to determine when `common/lzma/Alloc.c` is being compiled with the EE toolchain.

Another option could be replacing the use of the `_EE` macro with any target-specific macro set by the compiler such as `__R5900__` or other.